src/translators/decode.rs

```diff
@@ -78,32 +85,85 @@ pub fn decoding_key_from_secret(alg: &Algorithm, secret_string: &str) -> JWTResu
 
             let secret = slurp_file(&secret_string.chars().skip(1).collect::<String>());
 
-            match secret_string.ends_with(".pem") {
-                true => {
-                    DecodingKey::from_ec_pem(&secret).map_err(jsonwebtoken::errors::Error::into)
-                }
-                false => Ok(DecodingKey::from_ec_der(&secret)),
+            if secret_string.ends_with(".pem") {
+                DecodingKey::from_ec_pem(&secret).map_err(jsonwebtoken::errors::Error::into)
+            } else if secret_string.ends_with(".json") {
+                return decoding_key_from_jwks_secret(&secret, header);
+            } else {
+                Ok(DecodingKey::from_ec_der(&secret))
             }
         }
         Algorithm::EdDSA => {
             if !&secret_string.starts_with('@') {
-                return Err(JWTError::Internal(format!(
-                    "Secret for {alg:?} must be a file path starting with @",
-                )));
+                // allows to read JWKS from argument (e.g. output of 'curl https://auth.domain.com/jwks.json')
```

Comment:
if JWKS is nott supported for EdDSA why is this being read? I think this should be in the previous block where its missing

Reply:
Yes you are correct. My bad. https://github.com/mike-engel/jwt-cli/pull/310 should fix this